Update plugin to work with pytest 4.1.0

marcos-alfredo opened the pull request:
Hi,

The latest pytest update (4.1.0) removed the get_marker function, breaking the plugin. See comment on https://docs.pytest.org/en/latest/changelog.html for details.
The fix is just a single line replacement.

I wasn't sure if it was best to open an issue or this MR but since I fixed it for our project anyway, figure I'll send it your way and you can decide how to best go about it.

marcos-alfredo committed:
replace removed function with suggested alternative

get_marker function was removed, see this for reference https://github.com/pytest-dev/pytest/issues/4546

marcos-alfredo committed:
update plugin version and minimum pytest version

marcosalfredo-visma committed:
Merge pull request #1 from marcos-alfredo/patch-1

Patch 1

schettino72 commented on setup.py:
```diff
@@ -9,7 +9,7 @@
 setup(name = 'pytest-ignore-flaky',
       description = 'ignore failures from flaky tests (pytest plugin)',
-      version = '0.1.1',
+      version = '0.1.2',
```
Since this requires a major upgrade on py.test, we also need to bump our version number to `0.2.0`.
Or lets call it `1.0.0` :smile:

marcos-alfredo replied:
Sounds good, version changed.

schettino72 reviewed (changes requested):
Thanks. Pull-requests are always better than creating an issue...
Could you please also add an entry on `CHANGES` file?

Would you like to co-maintain this project? I am sending you github invitation.
Do you have a pypi account? I could also add you as a maintainer there so you can push a new release yourself...

marcosalfredo-visma committed:
change plugin version and update changelog

marcosalfredo-visma committed:
fix tests

marcos-alfredo commented:
Added the new entry to the CHANGES file and updated the version.

I've never worked with pypi but it should be an interesting experience, so sure.
Just created an account there, username marcos.alfredo

I also pushed a small change to fix the tests but it only worked for Python 3.4, I'll try to take a look later into why.

Edit: okay, quick look so for 2.7 I'll try freezing the version of doit. For Python 3.3, pytest doesn't list it as supported version, so maybe we should drop it? Also maybe add 3.5, 3.6 and 3.7 to the CI script. I'll check tomorrow.

schettino72 commented:
Are you still using python2.7? if not, just drop it...

Yes, supported versions should be 3.5 to 3.7.

schettino72 commented:
Cool. I have just added you as a maintainer on PyPI.
Feel free to add your name somewhere on README file, or create an AUTHORS file to give credit to yourself.
Also feel free to merge changes and publish on PyPI :)

marcosalfredo-visma committed:
add newer Python versions to CI and fix tests for 2.7

marcosalfredo-visma committed:
fixing pyflakes execution

marcos-alfredo commented:
I'm not using 2.7 but pytest still supports it and the plugin works, it was just a matter of fixing the tests on the CI pipeline.

It looks good now, I'll see about the authors thing.

After that should be good to merge and I'll figure out how to publish it.

marcosalfredo-visma committed:
change docs and setup in order to publish new version

marcosalfredo-visma committed:
add newline to end of file

marcos-alfredo commented:
Okay, I think this is it.
Added my name to the readme and the setup.py file.
Changed a few others things there that I think make sense.
I've tested how to publish the plugin on the PyPI test site, you can take a look here https://test.pypi.org/project/pytest-ignore-flaky/.

If you agree with all changes we can go ahead with the merge and I'll publish it to the main site.

schettino72 commented:
perfect. thanks.
if you could upload to PyPI, it would be great.
